### DISCUSSION TITLE
SWD targets not recognized

### DISCUSSION THREAD

**Discussion by chriskuku** [stlink-org/stlink#1240] [Q&A (resolved)] [answered]
I'm attaching a log I took today. I'm having very instable behaviour in recognizing SWD attached targets to nucleo and discovery boards which make it impossible to get a stable working setup on my Mac (mid 2014 Retina, macOS 11.6.5 BigSur). Very rarely it is possible to detect the target, but I haven't found a pattern yet, what the cause for the instability is.
I have powered the target from a Sabrent USB Hub. The measurable Voltage on the eval board is 5.02 V. OpenOCD reports 
3.23V on the target.

I'm clueless for the moment
[log.txt](https://github.com/stlink-org/stlink/files/8425257/log.txt)
.

**Comment by Nightwalker-87** [comment]
The only thing I can tell from the logfile is that your _libusb_ version is **1.0.26 rc1**, which is yet untested with this toolset as it is a pre-release. I can remember some issues with the _libusb_ library in the past with certain versions thereof, where we had to wait for the problem to be fixed in that external project. However this may or may not be related to your findings.

**Comment by chriskuku** [comment]
libusb 1.0.26 rc1 was intentional on advice of a libusb developer to test whether the segfault 11 also occurs on most recent version.

**Comment by chriskuku** [comment]
Just a question: how can I find out which checkout of stlink I did? Someone is asking me whether I have c4762e69 .

```
$ ./st-info --version
Failed to parse flash type or unrecognized flash type
v1.7.0-186-gc4762e6
```

BTW, why do I get this "Failed to parse..." note? A plain nucleo-F411RE board with jumpers CN2 closed is attached.

**Reply by Nightwalker-87** [reply]
The current checkout is displayed here: `v1.7.0-186-gc4762e6`

**Comment by chriskuku** [comment]
Hi, I'm coming back to the issue I have:

I have attached a Discovery board and a DIYMORE https://stm32-base.org/boards/STM32F407VGT6-diymore.html to the SWD-Interface.

```
$  ./st-info --version
Failed to parse flash type or unrecognized flash type
v1.7.0-186-gc4762e6
```

No clue for days now why the target isn't recognized. Connected SWCLK to PA14, SWDIO to PA13, Vcc externally powered.

Help greatly appreciated.


EDIT: False Alarm! Issue solved. It was hardware. Bad Berg connector crimping spot. Sorry for bothering.

```
$ ./st-info --probe
Failed to parse flash type or unrecognized flash type

detected chip_id parametres

# Device Type: STM32F4x5_F4x7
# Reference Manual: RM0090           // RM0090 (Rev. 2)
#
chip_id 0x413
flash_type 3
flash_size_reg 0x1fff7a22
flash_pagesize 0x4000
sram_size 0x30000
bootrom_base 0x1fff0000
bootrom_size 0x7800
option_base 0x40023c14
option_size 0x4
flags 2

Found 1 stlink programmers
  version:    V2J37S27
  serial:     0670FF505252836684442833
  flash:      1048576 (pagesize: 16384)
  sram:       196608
  chipid:     0x413

detected chip_id parametres

# Device Type: STM32F4x5_F4x7
# Reference Manual: RM0090           // RM0090 (Rev. 2)
#
chip_id 0x413
flash_type 3
flash_size_reg 0x1fff7a22
flash_pagesize 0x4000
sram_size 0x30000
bootrom_base 0x1fff0000
bootrom_size 0x7800
option_base 0x40023c14
option_size 0x4
flags 2

  dev-type:   STM32F4x5_F4x7

```
Nonetheless I'm wondering about the initial `Failed to parse flash type or unrecognized flash type`

Does it refer to the STLINK host?

**Reply by Nightwalker-87** [reply]
No, it refers to the target device. It seems to need some investigation from the code-side.

**Comment by Nightwalker-87** [comment, marked as answer]
This issue is now fixed in the current `nightly` branch (commit aee9a47e3551d3cac9449ef1cceed30ecdcff133).
It will merge into the `develop` branch soon.
